File: src/main/java/net/adventuregame/gameDecor/Tree.java
package net.adventuregame.gameDecor;

import com.chidozie.core.renderEngine.Loader;
import com.chidozie.core.renderEngine.OBJFileLoader;
import com.chidozie.core.textures.ModelTexture;
import net.adventuregame.entities.Entity;
import net.adventuregame.models.TexturedModel;
import org.joml.Vector3f;

public class Tree extends Entity {

    private static Loader loader = new Loader();
    private static TexturedModel treeModel = new TexturedModel(OBJFileLoader.loadOBJ("tree_legacy", loader), new ModelTexture(loader.loadGameTexture("tree")));
    static {
        ModelTexture texture = treeModel.getTexture();
        texture.setShineDamper(10);
        texture.setReflectivity(1);
    }

    public Tree(Vector3f position, float rotX, float rotY, float rotZ, float scale) {
        super(treeModel, position, rotX, rotY, rotZ, scale);
    }
}

File: src/main/kotlin/net/adventuregame/gameDecor/Tree.kt
package net.adventuregame.gameDecor

import com.adv.core.renderEngine.Loader
import com.adv.core.renderEngine.OBJFileLoader
import com.adv.core.textures.ModelTexture
import net.adventuregame.entity.Entity
import net.adventuregame.models.TexturedModel
import org.joml.Vector3f

class Tree(position: Vector3f, rotX: Float, rotY: Float, rotZ: Float, scale: Float) :
    Entity(position, rotX, rotY, rotZ, scale) {
    override fun GetModels(): List<TexturedModel> {
        return listOf(treeModel)
    }
    companion object {
        private val loader = Loader()
        private val treeModel =
            TexturedModel(OBJFileLoader.loadOBJ("tree_legacy", loader), ModelTexture(loader.loadGameTexture("tree")))

        init {
            val texture: ModelTexture = treeModel.texture!!
            texture.shineDamper = 10f
            texture.reflectivity = 1f
        }
    }
}
